sky/task.py

```diff
@@ -872,6 +884,10 @@ def num_nodes(self, num_nodes: Optional[int]) -> None:
                     f'num_nodes should be a positive int. Got: {num_nodes}')
         self._num_nodes = num_nodes
 
+    @property
+    def metadata(self) -> str:
+        return json.dumps(self._metadata)
```

Comment:
Hm, is there some reason we can't just have this just return a dict? That seems cleaner to me.
We can just update any upstream usages that expect a str to do the conversion themselves. Most usages seem like they want a dict anyway?

Reply:
Though looking at how it is being used, it seems like if we want it to be stored in json its better to keep it like this, since when we use it we always want it in json (we only use `task.metadata` in `set_pending` and `add_job` which both expect it to be json